lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -90,97 +139,346 @@ Function Remove-Repository{
     }
 }
 
+Function Get-PsModule {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [String]$Name,
+        [String]$RequiredVersion,
+        [String]$MinimumVersion,
+        [String]$MaximumVersion
+    )
+
+    $props = @{
+        'Exists' = $false
+        'Version' = ""
+        'Versions' = @("")
+    }
+    $ExistingModule = New-Object -TypeName psobject -Property $props
```

Comment:
We do try to avoid PSObject in our modules if there's no need for them.
@jborean93 I don't see it documented in https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_general_windows.html#windows-new-module-development, only `PSCustomObject` is mentioned. Should we add it ?